### DISCUSSION TITLE
resizing elements in the songselect skinning area doesnt change their "Anchor"/ some spaces can't be skinned.

### DISCUSSION THREAD

**Discussion by Inconsist** [ppy/osu#22210] [Q&A]
I've always been very annoyed at the fact that you can not really skin the songselect, becuase no matter where you put stuff, when you quit the editor everything shifts somewhere else anyways. 
I've found out that there are some deadzonese where you cant even place elements if you adjust the position with arrow keys(demonstarted in the video, the "unskinnable void" is colored white for reference).
So by trial and error I figured out that by resizing the elements where you want them to be instead fo draging them there, they shift a bit more predictable and the white space can infact be skinned. 
I demonstrate the extreme case of this in second vid. 

I assume both things are unintentional, that you can't skin this space aswell as the thing with the anchors, should I create an issue for this or is this stuff temporary and obsolete when skin editor gets worked on more?

https://user-images.githubusercontent.com/87947227/212520942-ede4e308-b44a-498d-8615-1f715188b063.mp4

https://user-images.githubusercontent.com/87947227/212520941-1670bd90-2726-4e11-b3eb-4e4a4cdd342b.mp4

---
As you can see in the second video the shifting gets more intense the further you are from the anchor, in my opinion this needs to be adressed somehow. Either make a universal anchor in the bottom left koordinate grid style or fix the shifting stuff with multiple anchors. Another thing that would be useful is if the skin editior would keep your original aspect ratio, this might also fix it and would make everything more predictable in behaviour and get rid of the need to shift stuff around alltogether.

**Comment by bdach** [comment]
The only sane way that I can see this working is by allowing elements to anchor to other elements rather than only the screen bounds. This will likely have to be done for gameplay skinning anyway.

**Reply by Inconsist** [reply]
And the idea to just keep the aspect ratio in the editor? I mean the same thing happens when you want to place something close to the edge of the screen(like a healthbar) in gameplay. It looks good in the editor and then shifts off the screen. Its feels really unpredictable.

**Reply by bdach** [reply]
What if you change resolution? move to windowed? etc. etc. Keeping aspect ratio in editor does nothing to fix that.

**Reply by Inconsist** [reply]
This is a valid point, but most people rarely change their resolution for every new session, so allowing them a way to skin their specific game once in a predictable way would seem like a good idea. Also im talking about aspect ratio, which even fewer people probably change regularly. If everything could anchor to the center it might also work.
